pkg/spoke/submarineragent/config_controller.go

```diff
@@ -181,6 +189,46 @@ func (c *submarinerConfigController) sync(ctx context.Context, syncCtx factory.S
 		return c.updateGatewayStatus(syncCtx.Recorder(), config)
 	}
 
+	if config.Status.ManagedClusterInfo.Platform == "GCP" {
+		if !helpers.IsSubmarinerEnvPrepared(c.configClient, config.Namespace, config.Name) {
+			cloudProvider, preparedErr := cloud.GetCloudProvider(c.restConfig, c.kubeClient, nil, c.dynamicClient,
+				c.hubKubeClient, c.eventRecorder, config.Status.ManagedClusterInfo, config)
+
+			if preparedErr != nil {
+				return preparedErr
+			} else {
+				preparedErr = cloudProvider.PrepareSubmarinerClusterEnv()
+			}
+
+			if preparedErr != nil {
+				return preparedErr
+			}
```

Comment:
We should be consistent with the `submarinerAgentController` which reports the error in the status condition.

Reply:
Done.